stwcs/wcsutil/altwcs.py

```diff
@@ -139,31 +145,15 @@ def archiveWCS(fname, ext, wcskey=" ", wcsname=" ", reusekey=False):
 
     log.setLevel('WARNING')
 
-    wkey = wkey.strip()
-
     for e in ext:
-        hdr = _getheader(f, e)
-        w = pywcs.WCS(hdr, f)
-        hwcs = w.to_header()
+        hwcs = wcs_from_key(f, e, from_key=' ', to_key=wkey)
 
         if hwcs is None:
             continue
 
-        if w.sip is not None:
-            for i in range(1, w.naxis + 1):
-                hwcs[f'CTYPE{i:d}'] += '-SIP'
+        hwcs.set(f'WCSNAME{wkey:.1s}', wname, 'Coordinate system title', before=0)
```

Comment:
Is this the official comment for WCSNAME?

Reply:
The best I could find: https://github.com/astropy/astropy/blob/bad46e22d293ca1c891bef0d0fae9e78c324eec5/cextern/wcslib/C/wcshdr.c#L992-L994